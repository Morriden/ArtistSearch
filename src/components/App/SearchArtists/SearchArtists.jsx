import React from 'react';
import PropTypes from 'prop-types';

const SearchArtists = ({ artists }) => {
  const artistsElements = artists.map(artist => (
    <li key={artist.id}>
      <div>
        {artist.name}
      </div>
    </li>
  ));

  return (
    <ul>
      {artistsElements}
    </ul>
  );
};

SearchArtists.propTypes = {
  artists: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default SearchArtists;
