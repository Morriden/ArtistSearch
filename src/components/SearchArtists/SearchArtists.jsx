import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const SearchArtists = ({ artists, handleSubmit, handleChange }) => {
  const artistsElements = artists.map(artist => (
   
    <li key={artist.id}>
      <Link to={`/${artist.id}`}>
        <div>
          {artist.name}
        </div>
      </Link>
    </li>

  ));

  return (
    <>
      <form onSubmit={handleSubmit}>
        <input type='text' onChange={handleChange}></input>
      </form>
      <ul>
        {artistsElements}
      </ul>
    </>
  );
};

SearchArtists.propTypes = {
  artists: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired
};

export default SearchArtists;
