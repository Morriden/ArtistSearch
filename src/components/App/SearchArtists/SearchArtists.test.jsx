import React from 'react';
import { shallow } from 'enzyme';
import SearchArtists from './SearchArtists';

describe('Search Artists snap shot tests', () => {
  it('shoud be a list of artists', () => {
    const wrapper = shallow(<SearchArtists artists={[{ name: 'artist 1', id: 1 }, { name: 'artist 2', id: 2 }, { name: 'artist 3', id: 3 }]}/>);
    
    expect(wrapper).toMatchSnapshot();
  });
});
