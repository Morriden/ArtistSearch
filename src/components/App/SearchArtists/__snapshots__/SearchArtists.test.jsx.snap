// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Artists snap shot tests shoud be a list of artists 1`] = `
<ul>
  <li
    key="1"
  >
    <div>
      artist 1
    </div>
  </li>
  <li
    key="2"
  >
    <div>
      artist 2
    </div>
  </li>
  <li
    key="3"
  >
    <div>
      artist 3
    </div>
  </li>
</ul>
`;
